### PR TITLE
[CIS-706] Fix custom channel path parameters

### PR DESCRIPTION
**Pull request by olejnjak**
Add tests and fix custom channel path parameters

### CODE REVIEW

File: Sources/StreamChat/Query/ChannelQuery_Tests.swift [hunk 1]
```diff
@@ -57,6 +57,32 @@ class ChannelQuery_Tests: XCTestCase {
...
+            imageURL: .unique(),
+            team: nil,
+            members: [.unique],
+            invites: [],
+            extraData: .defaultValue
+        ))
+        XCTAssertEqual(query.pathParameters, "custom_type")
+    }
+    
+    func test_pathParameters_with_customType_and_id() {
```
Comment (VojtaStavik) [inline Sources/StreamChat/Query/ChannelQuery_Tests.swift:76]: nit: we try to follow this pattern when naming tests

Comment (olejnjak) [reply Sources/StreamChat/Query/ChannelQuery_Tests.swift]: 👌

**Review by DominikBucher12 — APPROVED** [review]
LGTM 🎉 

Maybe just a nit I noticed in the ticket attachment - shouldn't we add to documentation inside code for `type` that there are only numbers, small letters and underscore allowed in the channel types? 

Or maybe this is unnecessary, but we could restrict the custom type to only allowed characters? Or at least add to documentation of `ChannelType`, I didn't know this until this PR 😅

**Review by DominikBucher12 — APPROVED** [review]
<3

File: Sources/StreamChat/Models/ChannelType.swift [hunk 2]
```diff
@@ -49,18 +53,34 @@ public enum ChannelType: Codable, Hashable {
...
         try container.encode(rawValue)
     }
+    
+    /// Check that custom types have valid `rawValue`
+    private static func assertCustomTypeValue(_ value: String) {
+        let allowedCharacters = CharacterSet.lowercaseLetters.union(CharacterSet(charactersIn: "0123456789_"))
+        let valueCharacters = CharacterSet(charactersIn: value)
+        log.assert(
+            valueCharacters.isSubset(of: allowedCharacters),
+            "Allowed characters for `ChannelType.custom` are small letters, numbers and underscore."
```
Comment (VojtaStavik) [inline Sources/StreamChat/Models/ChannelType.swift:83]: nit: Can we print the actual value in the error message? It will help with debugging

**Review by VojtaStavik — APPROVED** [review]
Looks good 👍 One nit fix and you can merge it :shipit:

**Pull request merged**